src/Tools/DateTimeComparator.php

```diff
@@ -5,57 +5,36 @@
 class DateTimeComparator
 {
     /**
-     * Get the greatest DateTime between the twice given as parameter
-     * If one DateTime is null, return the other
-     * If the twice DateTime are null, return null
+     * Get the greatest "DateTimeInterface" from the parameters
      */
-    public static function greatest(\DateTimeInterface $dateTime1 = null, \DateTimeInterface $dateTime2 = null)
+    public static function greatest(... $datetimes)
     {
-        // - If one DateTime is null, return the other date
-        // - If the twice DateTime are null, return null
-        if (\is_null($dateTime1) || \is_null($dateTime2)) {
-            return self::getNonNullableDateTime($dateTime1, $dateTime2);
+        $greatest = null;
+        foreach ($datetimes as $datetime) {
```

Comment:
Its more Code Style but I think the following is more friendly than indent several `if` statements

```
foreach ($datetimes as $datetime) {
    if (!$datetime instanceof \DateTimeInterface) {
        continue;
    }

    if ($datetime > $greatest || null === $greatest) {
        $greatest = $datetime;
    }
}
```